CI: Runtime Test (privileged) - TestDirectRoutingDevice

## CI failure

- Link: https://github.com/cilium/cilium/actions/runs/12889085928/job/35936266625

```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃   FAIL  package: github.com/cilium/cilium/pkg/datapath/tables   ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛

--- FAIL: TestDirectRoutingDevice (0.01s)

    logger.go:257: time=2025-01-21T15:05:52.403Z level=INFO source=/host/vendor/github.com/cilium/hive/hive.go:338 msg="Starting hive"
    logger.go:257: time=2025-01-21T15:05:52.403Z level=INFO source=/host/vendor/github.com/cilium/hive/hive.go:342 msg=Started duration=54.541µs
    direct_routing_device_test.go:133: 
        	Error Trace:	/host/pkg/datapath/tables/direct_routing_device_test.go:133
        	Error:      	Expected value not to be nil.
        	Test:       	TestDirectRoutingDevice
    logger.go:257: time=2025-01-21T15:05:52.413Z level=INFO source=/host/vendor/github.com/cilium/hive/hive.go:354 msg=Stopping
    logger.go:257: time=2025-01-21T15:05:52.413Z level=INFO source=/host/pkg/hive/health/provider.go:57 msg="health.job-module-status-metrics (rev=2)" module=health
```

--- a/pkg/datapath/tables/direct_routing_device_test.go
+++ b/pkg/datapath/tables/direct_routing_device_test.go
@@ -8,7 +8,6 @@ import (
 	"net"
 	"net/netip"
 	"testing"
-	"time"
 
 	"github.com/cilium/hive/cell"
 	"github.com/cilium/hive/hivetest"
@@ -62,20 +61,18 @@ func TestDirectRoutingDevice(t *testing.T) {
 		),
 	)
 
-	ctx := context.Background()
+	tctx := context.Background()
 	log := hivetest.Logger(t)
-	err := h.Start(log, ctx)
+	err := h.Start(log, tctx)
 	require.NoError(t, err)
-	t.Cleanup(func() { h.Stop(log, ctx) })
+	t.Cleanup(func() { h.Stop(log, tctx) })
 	require.NotNil(t, h)
 	require.NotNil(t, db)
 	require.NotNil(t, devicesTable)
 	require.NotNil(t, directRoutingDev)
 
 	// No devices - ensure that we don't get a device.
-	tctx, cancel := context.WithTimeout(ctx, time.Millisecond)
 	dev, _ := directRoutingDev.Get(tctx, db.ReadTxn())
-	cancel()
 	require.Nil(t, dev)
 
 	// Insert a device
@@ -85,19 +82,18 @@ func TestDirectRoutingDevice(t *testing.T) {
 		Name:     "direct0",
 		Selected: true,
 	}
-	devicesTable.Insert(txn, &want)
+	_, _, err = devicesTable.Insert(txn, &want)
+	require.NoError(t, err)
 	txn.Commit()
 
 	// And check that it's returned
-	tctx, cancel = context.WithTimeout(ctx, time.Millisecond)
 	got, watch := directRoutingDev.Get(tctx, db.ReadTxn())
-	cancel()
 	require.NotNil(t, got)
 	require.Equal(t, want.Name, got.Name)
 	select {
 	case <-watch:
 		t.Error("watch channel closed even though it should not")
-	case <-time.After(time.Millisecond):
+	default:
 	}
 
 	// Insert another device.
@@ -107,18 +103,17 @@ func TestDirectRoutingDevice(t *testing.T) {
 		Name:     "dummy0",
 		Selected: true,
 	}
-	devicesTable.Insert(txn, &dummyDev)
+	_, _, err = devicesTable.Insert(txn, &dummyDev)
+	require.NoError(t, err)
 	txn.Commit()
 
 	// Two selected devices - ensure that we don't get a device.
-	tctx, cancel = context.WithTimeout(ctx, time.Millisecond)
 	dev, watch = directRoutingDev.Get(tctx, db.ReadTxn())
-	cancel()
 	require.Nil(t, dev)
 	select {
 	case <-watch:
 		t.Error("watch channel closed even though it should not")
-	case <-time.After(time.Millisecond):
+	default:
 	}
 
 	// If one of the devices matches the K8s Node IP, it is returned.
@@ -127,15 +122,13 @@ func TestDirectRoutingDevice(t *testing.T) {
 			Addr: netipx.MustFromStdIP(testIP),
 		},
 	}
-	tctx, cancel = context.WithTimeout(ctx, time.Millisecond)
 	got, watch = directRoutingDev.Get(tctx, db.ReadTxn())
-	cancel()
 	require.NotNil(t, got)
 	require.Equal(t, want.Name, got.Name)
 	select {
 	case <-watch:
 		t.Error("watch channel closed even though it should not")
-	case <-time.After(time.Millisecond):
+	default:
 	}
 	want.Addrs = nil
 }
@@ -202,18 +195,18 @@ func TestDirectRoutingDevice_withConfig(t *testing.T) {
 				c.DirectRoutingDevice = tc.config
 			})
 
-			ctx := context.Background()
+			tctx := context.Background()
 			log := hivetest.Logger(t)
-			err := h.Start(log, ctx)
+			err := h.Start(log, tctx)
 			require.NoError(t, err)
-			t.Cleanup(func() { h.Stop(log, ctx) })
+			t.Cleanup(func() { h.Stop(log, tctx) })
 			require.NotNil(t, db)
 			require.NotNil(t, devicesTable)
 			require.NotNil(t, directRoutingDev)
 
 			// Insert devices
 			txn := db.WriteTxn(devicesTable)
-			devicesTable.Insert(txn, &Device{
+			_, _, err = devicesTable.Insert(txn, &Device{
 				Index:    10,
 				Name:     "dummy0",
 				Selected: true,
@@ -223,12 +216,12 @@ func TestDirectRoutingDevice_withConfig(t *testing.T) {
 					},
 				},
 			})
-			devicesTable.Insert(txn, &tc.want)
+			require.NoError(t, err)
+			_, _, err = devicesTable.Insert(txn, &tc.want)
+			require.NoError(t, err)
 			txn.Commit()
 
-			tctx, cancel := context.WithTimeout(ctx, time.Millisecond)
 			got, watch := directRoutingDev.Get(tctx, db.ReadTxn())
-			cancel()
 			if tc.wantErr == (got == nil) {
 				t.Errorf("wantErr %v but have err %v", tc.wantErr, err)
 			}
@@ -240,7 +233,7 @@ func TestDirectRoutingDevice_withConfig(t *testing.T) {
 			select {
 			case <-watch:
 				t.Error("watch channel closed even though it should not")
-			case <-time.After(time.Millisecond):
+			default:
 			}
 		})
 	}

